ngraph/core/src/op/strided_slice.cpp

```diff
@@ -174,28 +175,106 @@ void op::v1::StridedSlice::validate_and_infer_types()
+                                              strides_vector,
+                                              begin_axis_set,
+                                              end_axis_set,
+                                              new_axis_axis_set,
+                                              shrink_axis_axis_set,
+                                              ellipsis_axis_set);
+        else
+        {
+            // output shape dimensions of ss operation depend monotonically on the input shape
+            // dimensions: the larger the input dimensions the larger the output dimensions in order
+            // to use existing shape infer function of strided slice we split each dimension into
+            // lower and upper ones and put them into lower input shape and upper input shape
+            // correspondingly Example: data_shape = {{5, 9}, {0, 3}, -1, 5}, lower_shape = {5, 0,
+            // 0, 5}, upper_shape = {9, 3, max_int64, 5} With that we calculate two output shapes as
+            // if there were two strided slices but with different input shapes and combine them
+            // into one resulting output shape
+
+            auto lower_in_shape = PartialShape::dynamic(input_pshape.rank()),
+                 upper_in_shape = PartialShape::dynamic(input_pshape.rank());
+            for (auto i = 0; i < data_rank.get_length(); ++i)
+            {
+                lower_in_shape[i] = input_pshape[i].get_min_length();
+                upper_in_shape[i] = input_pshape[i].get_max_length();
+            }
+
+            auto lower_out_shape = infer_slice_shape(this,
+                                                     lower_in_shape,
+                                                     begin_vector,
+                                                     end_vector,
+                                                     strides_vector,
+                                                     begin_axis_set,
+                                                     end_axis_set,
+                                                     new_axis_axis_set,
+                                                     shrink_axis_axis_set,
+                                                     ellipsis_axis_set);
+
+            auto upper_out_shape = infer_slice_shape(this,
+                                                     upper_in_shape,
+                                                     begin_vector,
+                                                     end_vector,
+                                                     strides_vector,
+                                                     begin_axis_set,
+                                                     end_axis_set,
+                                                     new_axis_axis_set,
+                                                     shrink_axis_axis_set,
+                                                     ellipsis_axis_set);
+            NODE_VALIDATION_CHECK(this,
```

Comment:
If` lower_out_shape.rank().is_dynamic()` or `upper_out_shape.rank().is_dynamic`, can we set a dynamic rank to the output and not throw an exception?

Reply:
I can't make up a situation where it would happen and the rank check earlier didn't work:
```
if (data_rank.is_dynamic())    
{        
        set_output_type(0, get_input_element_type(0), PartialShape::dynamic());        
        return;   
}
```

Could you share your case?

Reply:
Ok, got it if
`data_rank.is_static()` then
`lower_in_shape, upper_in_shape` ranks are also static
and therefore we expect` lower_out_shape, upper_out_shape` are also static here.
Resolved.